core/src/com/unciv/logic/city/CityInfo.kt

```diff
@@ -128,7 +128,10 @@ class CityInfo {
     fun getWorkableTiles() = getTiles().filter { it in tilesInRange }
 
     fun isCapital() = cityConstructions.isBuilt("Palace")
-    fun isConnectedToCapital() = civInfo.citiesConnectedToCapital.contains(this)
+    fun isConnectedToCapital(connectionTypePredicate: (Set<String>) -> Boolean = {true}) =
+            civInfo.citiesConnectedToCapital[this]
+                    ?.let(connectionTypePredicate)
```

Comment:
Please don't use .let, it makes the function a lot less readable, especially with a ?: afterwards :/

Reply:
I'm used to let and others, but I'll follow conventions (my first in here :))